source/common/network/address_impl.cc

```diff
@@ -247,6 +257,12 @@ std::string Ipv6Instance::Ipv6Helper::makeFriendlyAddress() const {
   char str[INET6_ADDRSTRLEN];
   const char* ptr = inet_ntop(AF_INET6, &address_.sin6_addr, str, INET6_ADDRSTRLEN);
   ASSERT(str == ptr);
+  if (address_.sin6_scope_id != 0) {
+    // Note that here we don't use the `if_indextoname` that will give a more user friendly
+    // output just because in the past created a performance bottleneck if the machine had a
+    // lot of IPv6 Link local addresses.
```

Comment:
@yanavlasov can we us  `if_indextoname` is this is strictly new behavior and opt-in?

Reply:
Yes, it is safe for existing deployments. My concern was that it may come back to bite us when someone will use IPv6 link local addresses on machines with large number of IP interfaces.

As an aside the `makeFriendlyName` needs to be optimized (it is worth a few percentage points of CPU for configs with large number of EDS endpoints) and not even call inet_ntop for endpoint addresses that are already IP address strings and can just simply be copied into the friendly name. This will take care of resolving the interface name as well for the most part.

Reply:
Maybe I can try to take a look at that as part of [#23952](https://github.com/envoyproxy/envoy/issues/23952)